Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp

```diff
@@ -212,81 +228,156 @@ computePhiIGF ( amrex::MultiFab const & rho,
-#if     defined(AMREX_USE_CUDA)
-        heffte::fft3d_r2c<heffte::backend::cufft> fft
-#elif   defined(AMREX_USE_HIP)
-        heffte::fft3d_r2c<heffte::backend::rocfft> fft
+        // Number of cells in real space
+        const int nrx = local_box.length(0);
+        const int nry = local_box.length(1);
+        const int nrz = local_box.length(2);
+
+        // Number of cells in spectral space
+        const int nsx = c_local_box.length(0);
+        const int nsy = c_local_box.length(1);
+        const int nsz = c_local_box.length(2);
+
+        if(is_2d_slices){
+                        // Create many plans at once
+            int fft_size[] = {nry, nrx};
+            BL_PROFILE_VAR_START(timer_plans);
+            ablastr::math::anyfft::FFTplan forward_plan_rho = ablastr::math::anyfft::CreatePlanMany(
+                fft_size, tmp_rho[local_boxid].dataPtr(),
+                reinterpret_cast<ablastr::math::anyfft::Complex*>(tmp_rho_fft.dataPtr()),
+                ablastr::math::anyfft::direction::R2C, AMREX_SPACEDIM-1,
+                nrz, nullptr, 1, nrx*nry, nullptr, 1, nsx*nsy);
+            ablastr::math::anyfft::FFTplan forward_plan_G = ablastr::math::anyfft::CreatePlanMany(
+                fft_size, tmp_G[local_boxid].dataPtr(),
+                reinterpret_cast<ablastr::math::anyfft::Complex*>(tmp_G_fft.dataPtr()),
+                ablastr::math::anyfft::direction::R2C, AMREX_SPACEDIM-1,
+                nrz, nullptr, 1, nrx*nry, nullptr, 1, nsx*nsy);
+            ablastr::math::anyfft::FFTplan backward_plan = ablastr::math::anyfft::CreatePlanMany(
+                fft_size, tmp_G[local_boxid].dataPtr(),
+                reinterpret_cast<ablastr::math::anyfft::Complex*>(tmp_G_fft.dataPtr()),
+                ablastr::math::anyfft::direction::C2R, AMREX_SPACEDIM-1,
+                nsz, nullptr, 1, nsx*nsy, nullptr, 1, nrx*nry);
+            BL_PROFILE_VAR_STOP(timer_plans);
+
+            // Forward transforms of rho and G
+            BL_PROFILE_VAR_START(timer_ffts);
+            ablastr::math::anyfft::Execute(forward_plan_rho);
+            ablastr::math::anyfft::Execute(forward_plan_G);
+            BL_PROFILE_VAR_STOP(timer_ffts);
+
+            // Multiply tmp_G_fft and tmp_rho_fft in spectral space
+            // Store the result in-place in tmp_G_fft, to save memory
+            tmp_G_fft.template mult<amrex::RunOn::Device>(tmp_rho_fft, 0, 0, 1);
+            amrex::Gpu::streamSynchronize();
+
+            // Backtransform the product of FFT(rho) and FFT(G)
+            BL_PROFILE_VAR_START(timer_ffts);
+            ablastr::math::anyfft::Execute(backward_plan);
+            BL_PROFILE_VAR_STOP(timer_ffts);
+
+            // Remove plans
+            ablastr::math::anyfft::DestroyPlan(forward_plan_G);
+            ablastr::math::anyfft::DestroyPlan(forward_plan_rho);
+            ablastr::math::anyfft::DestroyPlan(backward_plan);
+
+            // Normalize, since (FFT + inverse FFT) results in a factor N but in 2d
```

Comment:
```suggestion
            // Normalize, since (FFT + inverse FFT) results in a factor N
```

Reply:
It wasn't clear to me what "but in 2d" meant here.

Reply:
What about increasing verbosity? 
```
// Normalize, since (FFT + inverse FFT) results in a factor N
// Note that we use nrx*nry instead of realspace_box.numPts() 
// because here FFTs are in 2D along x and y
```